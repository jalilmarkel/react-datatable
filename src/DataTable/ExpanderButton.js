import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';

const ButtonStyle = styled.button`
  outline: none;
  border: none;
  display: block;
  width: 42px;
  height: 42px;
  background-color: transparent;
  background-image: url(${props => props.theme.expander[props.expanded ? 'expandedButton' : 'collapsedButton']});
  background-position: center center;
  background-repeat: no-repeat;

  &:hover {
    cursor: pointer;
  }
`;

const ExpanderButton = ({ expanded, children, data, index, onToggled }) => {
  const handleToggle = e =>
    onToggled && onToggled(data, index, e);

  return (
    <ButtonStyle
      onClick={handleToggle}
      expanded={expanded}
    >
      {children}
    </ButtonStyle>
  );
};

ExpanderButton.propTypes = {
  data: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  expanded: PropTypes.bool,
  onToggled: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

ExpanderButton.defaultProps = {
  onToggled: null,
  children: null,
  expanded: false,
};

export default withTheme(ExpanderButton);
