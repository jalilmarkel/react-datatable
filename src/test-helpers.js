import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

