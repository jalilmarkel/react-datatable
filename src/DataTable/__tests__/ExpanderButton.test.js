import 'jest-styled-components';

import React from 'react';
import { shallowWithTheme } from '../../test-helpers';

import ExpanderButton from '../ExpanderButton';

test('component <ExpanderButton /> should render correctly', () => {
  const wrapper = shallowWithTheme(<ExpanderButton data={{}} index={0} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});
