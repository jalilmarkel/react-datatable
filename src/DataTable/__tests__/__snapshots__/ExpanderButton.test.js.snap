// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <ExpanderButton /> should render correctly 1`] = `
.c0 {
  outline: none;
  border: none;
  display: block;
  width: 42px;
  height: 42px;
  background-color: transparent;
  background-image: url(data:image/svg+xml,%3Csvg%20fill%3D%22%23757575%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20width%3D%2224%22%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%3E%0A%20%20%20%20%3Cpath%20d%3D%22M8.59%2016.34l4.58-4.59-4.58-4.59L10%205.75l6%206-6%206z%22/%3E%0A%20%20%20%20%3Cpath%20d%3D%22M0-.25h24v24H0z%22%20fill%3D%22none%22/%3E%0A%3C/svg%3E);
  background-position: center center;
  background-repeat: no-repeat;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
/>
`;
